pkg/metrics/metrics.go

```diff
@@ -361,3 +394,85 @@ func GetRhoamState(cr *integreatlyv1alpha1.RHMI) (RhoamState, error) {
 	}
 	return status, nil
 }
+
+func SloRemainingErrorBudget(route string, token config.Secret) (float32, prometheusv1.Warnings, error) {
+
+	sloMs := SloDays * 24 * 60 * 60 * 1000
+	slo001Ms := float64(sloMs) * 0.001
+
+	query := fmt.Sprintf("%[1]v - (sum_over_time((clamp_max(sum(ALERTS{alertstate=\"firing\", severity=\"critical\", product=\"rhoam\"}), 1))[%[2]vd:10m]) * (10 * 60 * 1000))  or vector(%[1]v)", slo001Ms, SloDays)
+
+	value, warnings, err := vectorQuery(route, token, query)
+	if err != nil {
+		err = fmt.Errorf("slo remaining error budget : %s", err)
+	}
+
+	return value, warnings, err
+
+}
+
+func SloPercentile(route string, token config.Secret) (float32, prometheusv1.Warnings, error) {
+
+	query := fmt.Sprintf("clamp_max(sum_over_time((clamp_max(sum(absent(ALERTS{alertstate=\"firing\", severity=\"critical\", product=\"rhoam\"})), 1))[%[1]vd:10m]) / (%[1]v * 24 * 6) > 0, 1)", SloDays)
+
+	value, warnings, err := vectorQuery(route, token, query)
+	if err != nil {
+		err = fmt.Errorf("slo percentile : %s", err)
+	}
+
+	return value, warnings, err
+}
+
+func GetApiClient(route string, token config.Secret) (prometheusv1.API, error) {
+	client, err := api.NewClient(api.Config{
+		Address:      route,
+		RoundTripper: config.NewAuthorizationCredentialsRoundTripper("Bearer", token, api.DefaultRoundTripper),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	v1api := prometheusv1.NewAPI(client)
+	return v1api, nil
+}
+
+func vectorQuery(route string, token config.Secret, query string) (float32, prometheusv1.Warnings, error) {
+	v1api, err := GetApiClient(route, token)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	result, warnings, err := v1api.Query(ctx, query, time.Now())
+	if err != nil {
+		return 0, nil, err
+	}
+
+	resultValue := result.(model.Vector)
+
+	if len(resultValue) == 1 {
+		value := float32(resultValue[0].Value)
+		return value, warnings, nil
+	} else {
+		return 0, warnings, fmt.Errorf("vector models length is unexpected: Expected 1; Got %v", len(resultValue))
+	}
```

Comment:
```suggestion
	} 
	
	return 0, warnings, fmt.Errorf("vector models length is unexpected: Expected 1; Got %v", len(resultValue))
```
nit: reduntant `else`

Reply:
I am going to except this but I am in two minds about it. One of the items I address in the commit "MGDAPI-4807 : golangci-lint refactor" was caused by try not to have the redundant else. The use of the else seem to be more explicit in what should happen next while not using it gives the impression that more might be allowed to happen afterwards. At some point I would like to open a discussion with the wider group on this pattern. For now I am happy to except it.